### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by akalsi87** [Cloaked9000/frnetlib#13]
This may sound naive, and I apologize if it is.

I am looking for a simple cross platform socket abstraction library, and everything on the README made sense to me.

Then I saw this:
https://github.com/Cloaked9000/frnetlib/blob/master/include/frnetlib/Socket.h#L50-L53

I'm wondering what mutexes are used for here? Why does a socket need a mutex instead of it being the responsibility of the user of the Socket?

**Comment by Cloaked9000** [comment]
You're absolutely correct, the socket shouldn't really be doing that itself. 

It's a bit of a remnant from the original purpose of the library. Originally, I wrote this library to replace the networking code within another project, which relied on the sockets being capable of accepting multiple simultaneous messages from different threads. 

The other project no longer depends on this behavior and so it's quite redundant, I've removed it: https://github.com/Cloaked9000/frnetlib/commit/ff25d11089547c78fd1a49fc6355c3c10434b216
